lib/model/message_list.dart

```diff
@@ -528,9 +567,21 @@ class MessageListView with ChangeNotifier, _MessageSequence {
 
       _insertAllMessages(0, fetchedMessages);
       _haveOldest = result.foundOldest;
+      _fetchOlderCooldownBackoffMachine = null;
```

Comment:
nit: let's put this line that ends a backoff series right next to the logic that starts a backoff — as an `else` to the `if (hasFetchError)`.

I believe that's equivalent to this way. And that way it's somewhat easier to see how this whole cooldown state machine operates.